rest/src/server/bootstrapper.js

```diff
@@ -93,23 +94,59 @@ module.exports = {
 
 	/**
 	 * Creates a REST api server.
-	 * @param {array} crossDomainConfig Configuration related to access control, contains allowed host and HTTP methods.
+	 * @param {array} config Application configuration
 	 * @param {object} formatters Formatters to use for formatting responses.
 	 * @param {object} throttlingConfig Throttling configuration parameters, if not provided throttling won't be enabled.
 	 * @returns {object} Server.
 	 */
-	createServer: (crossDomainConfig, formatters, throttlingConfig) => {
-		// create the server using a custom formatter
-		const server = restify.createServer({
+	createServer: (config, formatters, throttlingConfig) => {
+		let serverOptions = {
 			name: '', // disable server header in response
 			formatters: {
 				'application/json': formatters.json
 			}
-		});
+		};
+		if (!config.protocol)
+			winston.warn('Protocol is not configured explicitly in the configuration, defaulting to HTTPS.');
+
+		const protocol = config.protocol ? config.protocol : 'HTTPS';
+		winston.warn(`Using protocol: ${protocol}`);
+		// create http or https server based on the configuration and default will be https
+		if (config && 'HTTPS' === protocol) {
+			let sslKey; let
+				sslCertificate;
+			if (!config.sslKeyPath) {
+				throw new Error('Server default is HTTPS but no SSL Key found, '
+				+ '\'sslKeyPath\' property in the configuration must be provided.');
+			} else {
+				try {
+					sslKey = fs.readFileSync(config.sslKeyPath);
+				} catch (err) {
+					throw new Error(`SSL Key file cannot be found at the path: ${config.sslKeyPath}`);
+				}
```

Comment:
this catch does not seem to add much value. since it is throwing a new error distinct from the original one, the original callstack is being lost, which might make diagnosing problems harder. [e.g. maybe the root cause is a file perm issue]